ui/packages/consul-ui/tests/integration/adapters/kv-test.js

```diff
@@ -132,20 +138,20 @@ module('Integration | Adapter | kv', function(hooks) {
     const adapter = this.owner.lookup('adapter:kv');
     const client = this.owner.lookup('service:client/http');
     const request = client.url.bind(client);
-    assert.throws(function() {
+    assert.rejects(
       adapter.requestForQuery(request, {
         dc: dc,
-      });
-    });
+      })
+    );
   });
   test("requestForQueryRecord throws if you don't specify an id", function(assert) {
     const adapter = this.owner.lookup('adapter:kv');
     const client = this.owner.lookup('service:client/http');
     const request = client.url.bind(client);
-    assert.throws(function() {
+    assert.rejects(
       adapter.requestForQueryRecord(request, {
         dc: dc,
-      });
-    });
+      })
+    );
```

Comment:
As explained up above, I added an `async` keyword to some adapter methods (but only the ones necessary for this PR). This meant that the mock things we create to test things also need to by 'promised'. So the majority of changes to the adapter tests here do exactly the same as this change, again and again and again.